### PR TITLE
Fix Number.isInteger(), etc. in old type inference

### PR DESCRIPTION
**Pull request by Dominator008**
As suggested in #1162.

[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1552)

### CODE REVIEW

**Comment by dimvar** [comment]
Assigning to self. Depending on what we decide for #1545, we'll revisit this as well.

**Comment by dimvar** [comment]
So, turns out OTI doesn't restrict the type of an untyped variable if it's passed to a typed function. Since it generally doesn't do this, we shouldn't just special-case these functions to do it.

http://closure-compiler-debugger.appspot.com/#input0%3Dfunction%2520f(%252F**%2520number%2520*%252F%2520x)%2520%257B%257D%250A%250Afunction%2520g(x)%2520%257B%250A%2520%2520f(x)%253B%250A%2520%2520var%2520%252F**%2520string%2520*%252F%2520s%2520%253D%2520x%253B%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by concavelenz** [comment]
Dimvar@ we do this with a number of specific cases: goog.asserts methods, goog.isX method.  isInteger is a natural extensions of this

**Comment by Dominator008** [comment]
@concavelenz LG?

**Comment by concavelenz** [comment]
LGTM

File: test/com/google/javascript/jscomp/TypeInferenceTest.java [hunk 1]
```diff
@@ -1516,6 +1516,40 @@ public void testNotIsArray() {
...
+    assuming("x", createUnionType(ARRAY_TYPE, NUMBER_TYPE));
+    inFunction("goog.asserts.assert(Number.isSafeInteger(x));");
+    verify("x", NUMBER_TYPE);
+
+    assuming("x", createUnionType(ARRAY_TYPE, NUMBER_TYPE));
+    inFunction("goog.asserts.assert(!Number.isFinite(x));");
+    verify("x", NUMBER_TYPE);
+
+    assuming("x", createUnionType(ARRAY_TYPE, NUMBER_TYPE));
+    inFunction("goog.asserts.assert(!Number.isInteger(x));");
```
Comment (blickly) [inline test/com/google/javascript/jscomp/TypeInferenceTest.java]: Not sure I follow here. If something is not an integer, how do we know it's a number type?

Comment (Dominator008) [inline test/com/google/javascript/jscomp/TypeInferenceTest.java]: I think the intention is that `x` should always be a number type: https://github.com/google/closure-compiler/blob/master/externs/es6.js#L1525-L1531.

Comment (blickly) [inline test/com/google/javascript/jscomp/TypeInferenceTest.java]: Then what makes the Numer.isXXX methods different than any other extern definitions?

Comment (Dominator008) [inline test/com/google/javascript/jscomp/TypeInferenceTest.java]: I wrote this before seeing that the externs are already stricter than the specs. I think this change is unnecessary then.

**Comment by Dominator008** [comment]
I wrote this before seeing that the externs are already stricter than the specs. I think this change is unnecessary then.

**Pull request closed without merging**